pkg/ansible/paramconv/paramconv.go

```diff
@@ -84,6 +91,38 @@ func ToSnake(s string) string {
 	bits := []string{}
 	n := ""
 	iReal := -1
+	count := 0
+	length := len(s)
+	for i, v := range s {
+		if i+1 < length {
+			next := s[i+1]
+			count = count + 1
+			if (v >= 'A' && v <= 'Z' && next >= 'a' && next <= 'z') || (v >= 'a' && v <= 'z' && next >= 'A' && next <= 'Z') {
+				if next >= 'A' && next <= 'Z' {
+					count = 0
+				} else if next >= 'a' && next <= 'z' && next != 's' {
+					// fmt.Printf("This is the slice: %s", s[i-count+1:i+1])
+					if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+1])]; ok && i-count+1 == 0 {
+						s = s[0:i+1] + "_" + s[i+2:]
+					} else if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+1])]; ok {
+						s = s[0:i-count+1] + "_" + s[i-count+1:i+1] + "_" + s[i+1:]
+					}
+					// fmt.Printf("This is the final string: %s", s)
+				} else if next >= 'a' && next <= 'z' && next == 's' {
+					if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+2])]; ok && i+2 == length {
+						s = s[0:i-count+1] + "_" + s[i-count+1:i+1] + "S"
+						// fmt.Printf("\n this si the S : %s\n", s)
+					} else if _, ok := wordMapping[strings.ToLower(s[i-count+1:i+2])]; ok && i-count+1 == 0 {
+						s = s[0:i+1] + "S" + "_" + s[i+2:]
+					} else {
+						s = s[0:i-count+1] + "_" + s[i-count+1:i+1] + "S" + "_" + s[i+2:]
+					}
+				}
+
+			}
+		}
+
+	}
```

Comment:
@Venkat19967 My first thought initially was wondering if there was a way to combine this loop with the one below, but I see you are actually modifying the string `s` that is used in the loop below. Given that I would break this out into a function, something like `preprocessWordMappings(...) string`. Not sure what parameters it would take but it should return a string.

So then in this method the above code would be replaced by the method call:

`s = preprocessWordMappings(...)`

I think that will make it easier to understand. Then add a comment above explaining what you are doing in the method.